### PR TITLE
Fix: wso2/product-ei#2557

### PR DESCRIPTION
**Pull request by SangeethKarunaratne**
Fixed dependency download with embeded Maven.

Fixes: https://github.com/wso2/product-ei/issues/2557

### CODE REVIEW

File: plugins/org.wso2.developerstudio.eclipse.gmf.esb.edit/src-gen/org/wso2/developerstudio/eclipse/gmf/esb/parts/forms/DependencyProvider.java [hunk 1]
```diff
@@ -1166,39 +1174,71 @@ private void showMessage(String msg) {
...
+                    new SubProgressMonitor(monitor, 1));
+            request.setGoals(goals);
+            request.setShowErrors(true);
+            request.setUpdateSnapshots(true);
+            result = maven.execute(request, new SubProgressMonitor(monitor, 1));
+
+            return result;
+
+        } catch (Throwable e) {
+            e.printStackTrace();
```
Comment (dilee) [inline plugins/org.wso2.developerstudio.eclipse.gmf.esb.edit/src-gen/org/wso2/developerstudio/eclipse/gmf/esb/parts/forms/DependencyProvider.java]: Use a logger and log the message along with the trace.

File: plugins/org.wso2.developerstudio.eclipse.gmf.esb.edit/src-gen/org/wso2/developerstudio/eclipse/gmf/esb/parts/forms/DependencyProvider.java [hunk 2]
```diff
@@ -1166,39 +1174,71 @@ private void showMessage(String msg) {
...
+                for (File file : sourceFiles) {
+                    if (file.getName().endsWith(".jar")) {
+                        FileUtils.copyFile(file, new File(getWorkingDirectory() + File.separator + microesbLibPath
+                                + File.separator + file.getName()));
+                    }
                 }
             }
 
+        } catch (java.net.UnknownHostException e1) {
+            e1.printStackTrace();
```
Comment (dilee) [inline plugins/org.wso2.developerstudio.eclipse.gmf.esb.edit/src-gen/org/wso2/developerstudio/eclipse/gmf/esb/parts/forms/DependencyProvider.java]: Let's use a logger.

File: plugins/org.wso2.developerstudio.eclipse.gmf.esb.edit/src-gen/org/wso2/developerstudio/eclipse/gmf/esb/parts/forms/DependencyProvider.java [hunk 3]
```diff
@@ -1166,39 +1174,71 @@ private void showMessage(String msg) {
...
-                        MessageDialog.openInformation(dialogShell, "An Error has occoured",
-                                "An error has occured while downloading the Driver"
-                                        + "\nMake sure maven is install in your computer");
-                    }
-                });
-            }
-        } finally {
-            p.destroy();
+        } catch (Throwable e) {
+            e.printStackTrace();
```
Comment (dilee) [inline plugins/org.wso2.developerstudio.eclipse.gmf.esb.edit/src-gen/org/wso2/developerstudio/eclipse/gmf/esb/parts/forms/DependencyProvider.java]: Use a logger.

File: plugins/org.wso2.developerstudio.eclipse.gmf.esb.edit/src-gen/org/wso2/developerstudio/eclipse/gmf/esb/parts/forms/DependencyProvider.java [hunk 4]
```diff
@@ -1467,61 +1484,49 @@ public void run(IProgressMonitor monitor) throws InvocationTargetException, Inte
...
-                                        + "\n\nThe downloaded driver is located in the dependencies directory."
-                                        + "\nNote that downloaded drivers are not deployed with the CApp.");
+                        remove(dependencyDir + File.separator + "target");
+                        remove(dependencyDir + File.separator + "pom.xml");
+                        MessageDialog.openInformation(shell, "An Error has occoured",
+                                "An error occurred when downloading the driver."
+                                        + "\nMake sure that you are connected to the internet.");
                     }
                 });
+            } catch (InterruptedException e) {
```
Comment (dilee) [inline plugins/org.wso2.developerstudio.eclipse.gmf.esb.edit/src-gen/org/wso2/developerstudio/eclipse/gmf/esb/parts/forms/DependencyProvider.java]: Handle the exception. Use a logger.

File: plugins/org.wso2.developerstudio.eclipse.gmf.esb.edit/src-gen/org/wso2/developerstudio/eclipse/gmf/esb/parts/forms/DependencyProvider.java [hunk 5]
```diff
@@ -1467,61 +1484,49 @@ public void run(IProgressMonitor monitor) throws InvocationTargetException, Inte
...
+            } catch (Throwable e) {
+                e.printStackTrace();
```
Comment (dilee) [inline plugins/org.wso2.developerstudio.eclipse.gmf.esb.edit/src-gen/org/wso2/developerstudio/eclipse/gmf/esb/parts/forms/DependencyProvider.java]: Use a logger.

File: plugins/org.wso2.developerstudio.eclipse.gmf.esb.edit/src-gen/org/wso2/developerstudio/eclipse/gmf/esb/parts/forms/DependencyProvider.java [hunk 6]
```diff
@@ -1467,61 +1484,49 @@ public void run(IProgressMonitor monitor) throws InvocationTargetException, Inte
...
-                remove(dependencyDir + File.separator + "pom.xml");
+            } catch (java.net.UnknownHostException e1) {
                 Display.getDefault().syncExec(new Runnable() {
                     public void run() {
-                        MessageDialog.openInformation(shell, "Success",
-                                "You have successfully downloaded the driver."
-                                        + "\n\nThe downloaded driver is located in the dependencies directory."
-                                        + "\nNote that downloaded drivers are not deployed with the CApp.");
+                        remove(dependencyDir + File.separator + "target");
+                        remove(dependencyDir + File.separator + "pom.xml");
```
Comment (dilee) [inline plugins/org.wso2.developerstudio.eclipse.gmf.esb.edit/src-gen/org/wso2/developerstudio/eclipse/gmf/esb/parts/forms/DependencyProvider.java]: Let's use constants for strings

File: plugins/org.wso2.developerstudio.eclipse.gmf.esb.edit/src-gen/org/wso2/developerstudio/eclipse/gmf/esb/parts/forms/DependencyProvider.java [hunk 7]
```diff
@@ -1467,61 +1484,49 @@ public void run(IProgressMonitor monitor) throws InvocationTargetException, Inte
...
                         public void run() {
-                            remove(dependencyDir + "/target");
-                            remove(dependencyDir + "/pom.xml");
-                            MessageDialog.openInformation(shell, "An Error has occoured",
-                                    "An error occurred when downloading the driver."
-                                            + "\nMake sure that you have installed Apache Maven.");
+                            MessageDialog.openInformation(shell, "Success",
+                                    "You have successfully downloaded the driver."
+                                            + "\n\nThe downloaded driver is located in the dependencies directory."
+                                            + "\nNote that downloaded drivers are not deployed with the CApp.");
```
Comment (dakshika) [inline plugins/org.wso2.developerstudio.eclipse.gmf.esb.edit/src-gen/org/wso2/developerstudio/eclipse/gmf/esb/parts/forms/DependencyProvider.java]: IMO we need to say "You need to copy jars on dependencies directory to <PRODUCT_HOME>/lib"

File: plugins/org.wso2.developerstudio.eclipse.gmf.esb.edit/src-gen/org/wso2/developerstudio/eclipse/gmf/esb/parts/forms/DependencyProvider.java (hunk 1 above)
Comment (SangeethKarunaratne) [reply plugins/org.wso2.developerstudio.eclipse.gmf.esb.edit/src-gen/org/wso2/developerstudio/eclipse/gmf/esb/parts/forms/DependencyProvider.java]: fixed

File: plugins/org.wso2.developerstudio.eclipse.gmf.esb.edit/src-gen/org/wso2/developerstudio/eclipse/gmf/esb/parts/forms/DependencyProvider.java (hunk 2 above)
Comment (SangeethKarunaratne) [reply plugins/org.wso2.developerstudio.eclipse.gmf.esb.edit/src-gen/org/wso2/developerstudio/eclipse/gmf/esb/parts/forms/DependencyProvider.java]: fixed

File: plugins/org.wso2.developerstudio.eclipse.gmf.esb.edit/src-gen/org/wso2/developerstudio/eclipse/gmf/esb/parts/forms/DependencyProvider.java (hunk 3 above)
Comment (SangeethKarunaratne) [reply plugins/org.wso2.developerstudio.eclipse.gmf.esb.edit/src-gen/org/wso2/developerstudio/eclipse/gmf/esb/parts/forms/DependencyProvider.java]: fixed

File: plugins/org.wso2.developerstudio.eclipse.gmf.esb.edit/src-gen/org/wso2/developerstudio/eclipse/gmf/esb/parts/forms/DependencyProvider.java (hunk 4 above)
Comment (SangeethKarunaratne) [reply plugins/org.wso2.developerstudio.eclipse.gmf.esb.edit/src-gen/org/wso2/developerstudio/eclipse/gmf/esb/parts/forms/DependencyProvider.java]: fixed

File: plugins/org.wso2.developerstudio.eclipse.gmf.esb.edit/src-gen/org/wso2/developerstudio/eclipse/gmf/esb/parts/forms/DependencyProvider.java (hunk 5 above)
Comment (SangeethKarunaratne) [reply plugins/org.wso2.developerstudio.eclipse.gmf.esb.edit/src-gen/org/wso2/developerstudio/eclipse/gmf/esb/parts/forms/DependencyProvider.java]: fixed

File: plugins/org.wso2.developerstudio.eclipse.gmf.esb.edit/src-gen/org/wso2/developerstudio/eclipse/gmf/esb/parts/forms/DependencyProvider.java (hunk 7 above)
Comment (SangeethKarunaratne) [reply plugins/org.wso2.developerstudio.eclipse.gmf.esb.edit/src-gen/org/wso2/developerstudio/eclipse/gmf/esb/parts/forms/DependencyProvider.java]: fixed

File: plugins/org.wso2.developerstudio.eclipse.gmf.esb.edit/src-gen/org/wso2/developerstudio/eclipse/gmf/esb/parts/forms/DependencyProvider.java (hunk 6 above)
Comment (SangeethKarunaratne) [reply plugins/org.wso2.developerstudio.eclipse.gmf.esb.edit/src-gen/org/wso2/developerstudio/eclipse/gmf/esb/parts/forms/DependencyProvider.java]: fixed

File: plugins/org.wso2.developerstudio.eclipse.gmf.esb.edit/src-gen/org/wso2/developerstudio/eclipse/gmf/esb/presentation/DependencyProvider.java [hunk 8]
```diff
@@ -1451,78 +1486,94 @@ public int getDatabaseIndex(String arr[], String dbType) {
...
-                                MessageDialog.openInformation(shell, "Cancelled",
-                                        "JDBC driver download operation was cancelled");
+                                remove(dependencyDir + File.separator + TARGET);
+                                MessageDialog.openInformation(shell, "An Error has occoured",
+                                        "An error occurred when downloading the driver."
+                                                + "\nMake sure that you are connected to the internet.");
                             }
                         });
-                        throw new InterruptedException("JDBC driver download operation was cancelled");
+                        log.info("Error while downloading depenency", e);
```
Comment (dakshika) [inline plugins/org.wso2.developerstudio.eclipse.gmf.esb.edit/src-gen/org/wso2/developerstudio/eclipse/gmf/esb/presentation/DependencyProvider.java]: incorrect log type

Comment (SangeethKarunaratne) [reply plugins/org.wso2.developerstudio.eclipse.gmf.esb.edit/src-gen/org/wso2/developerstudio/eclipse/gmf/esb/presentation/DependencyProvider.java]: fixed

**Pull request merged**